Library/Homebrew/tap_constants.rb

```diff
@@ -1,9 +1,11 @@
+# allows tap names to start with https://www.github.com/, which is then stripped off
+HOMEBREW_TAP_URL_REGEX = %r{(?:(?:(?:https?://)?(?:www\.)?)?github\.com/)?}
 # match expressions when taps are given as ARGS, e.g. someuser/sometap
-HOMEBREW_TAP_ARGS_REGEX = %r{^([\w-]+)/(homebrew-)?([\w-]+)$}
+HOMEBREW_TAP_ARGS_REGEX = %r{^#{HOMEBREW_TAP_URL_REGEX}([\w-]+)/(homebrew-)?([\w-]+)$}
 # match taps' formulae, e.g. someuser/sometap/someformula
-HOMEBREW_TAP_FORMULA_REGEX = %r{^([\w-]+)/([\w-]+)/([\w+-.]+)$}
+HOMEBREW_TAP_FORMULA_REGEX = %r{^#{HOMEBREW_TAP_URL_REGEX}([\w-]+)/([\w-]+)/([\w+-.]+)$}
```

Comment:
We have defined tap formula's full-qualified name as `user/repo/formula` since we begin to support tap. And such format is used almost everywhere in the core and some external commands. Therefore, we cannot change it.